build/webpack.client.config.js

```diff
@@ -19,26 +19,31 @@ const config = merge(base, {
       'process.env.NODE_ENV': JSON.stringify(process.env.NODE_ENV || 'development'),
       'process.env.VUE_ENV': '"client"'
     }),
-    // extract vendor chunks for better caching
-    new webpack.optimize.CommonsChunkPlugin({
-      name: 'vendor',
-      minChunks: function (module) {
-        // a module is extracted into the vendor chunk if...
-        return (
-          // it's inside node_modules
-          /node_modules/.test(module.context) &&
-          // and not a CSS file (due to extract-text-webpack-plugin limitation)
-          !/\.css$/.test(module.request)
-        )
-      }
-    }),
-    // extract webpack runtime & manifest to avoid vendor chunk hash changing
-    // on every build.
-    new webpack.optimize.CommonsChunkPlugin({
-      name: 'manifest'
-    }),
+
     new VueSSRClientPlugin()
-  ]
+  ],
+  optimization:{
+        runtimeChunk: {
+            name: "manifest"
+        },
+      // extract webpack runtime & manifest to avoid vendor chunk hash changing
+      // on every build.
+      // extract vendor chunks for better caching
+        splitChunks:{
+            chunks:"initial",
+            cacheGroups: {
+                vendors: function (module) {
```

Comment:
I tried your code using webpack 4.23.1 and it fails with a "TypeError: cacheGroup.getName is not a function"
The correct form for cacheGroups entries seems different (see  https://webpack.js.org/plugins/split-chunks-plugin/#splitchunks-cachegroups)

```
  vendors: {
    test: function(module) {
```